Raven.Database/Queries/MoreLikeThisQueryRunner.cs

```diff
@@ -232,12 +243,32 @@ private JsonDocument[] GetJsonDocuments(MoreLikeThisQuery parameters, IndexSearc
                 .Select(hit => new JsonDocument
                 {
                     DataAsJson = Index.CreateDocumentFromFields(searcher.Doc(hit.Doc),
-                                                                new FieldsToFetch(fields, false, index.IsMapReduce ? Constants.ReduceKeyFieldName : Constants.DocumentIdFieldName)),
+                        new FieldsToFetch(fields, false, index.IsMapReduce ? Constants.ReduceKeyFieldName : Constants.DocumentIdFieldName)),
                     Etag = etag
                 })
                 .ToArray();
         }
 
+        private JsonDocument[] GetJsonDocuments(IndexSearcher searcher, Index index, string indexName, ScoreDoc[] scoreDocs)
+        {
+            if (scoreDocs.Any())
+            {
+                // Since we don't have a document we get the fields from the first hit
+                var fields = searcher.Doc(scoreDocs.First().Doc).GetFields().Cast<AbstractField>().Select(x => x.Name).Distinct().ToArray();
```

Comment:
They have a difference: __dummy_force_index_write vs __reduced_val

I think we should have the __reduced_val there.